homeassistant/components/sensor/lacrosse.py

```diff
@@ -0,0 +1,217 @@
+from homeassistant.core import callback
+from homeassistant.components.sensor import (ENTITY_ID_FORMAT, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_STOP, CONF_DEVICE, CONF_NAME, CONF_ID,
+    CONF_SENSORS, CONF_TYPE, TEMP_CELSIUS)
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util import dt as dt_util
+
+REQUIREMENTS = ['pylacrosse==0.2.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BAUD = 'baud'
+CONF_EXPIRE_AFTER = 'expire_after'
+
+DEFAULT_DEVICE = '/dev/ttyUSB0'
+DEFAULT_BAUD = '57600'
+DEFAULT_EXPIRE_AFTER = 300
+
+TYPES = ['battery', 'humidity', 'temperature']
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Required(CONF_ID): cv.positive_int,
+    vol.Required(CONF_TYPE): vol.In(TYPES),
+    vol.Optional(CONF_EXPIRE_AFTER): cv.positive_int,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSORS): vol.Schema({cv.slug: SENSOR_SCHEMA}),
+    vol.Optional(CONF_BAUD, default=DEFAULT_BAUD): cv.string,
+    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the LaCrosse sensors."""
+    import pylacrosse
+    from serial import SerialException
+
+    usb_device = config.get(CONF_DEVICE)
+    baud = int(config.get(CONF_BAUD))
+    expire_after = config.get(CONF_EXPIRE_AFTER)
+
+    _LOGGER.debug("%s %s", usb_device, baud)
+
+    try:
+        lacrosse = pylacrosse.LaCrosse(usb_device, baud)
+        lacrosse.open()
+    except SerialException as exc:
+        _LOGGER.warning("Unable to open serial port: %s", exc)
+        return False
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, lacrosse.close)
+
+    sensors = []
+    for device, device_config in config[CONF_SENSORS].items():
+        _LOGGER.debug("%s %s", device, device_config)
+
+        typ = device_config.get(CONF_TYPE)
+        sensor_class = TYPE_CLASSES[typ]
+        name = device_config.get(CONF_NAME, device)
+
+        sensors.append(
+            sensor_class(
+                hass, lacrosse, device, name, expire_after, device_config
+            )
+        )
+
+    add_devices(sensors)
+
+
+class LaCrosseSensor(Entity):
+    """Implementation of a Lacrosse sensor."""
+
+    _temperature = None
+    _humidity = None
+    _low_battery = None
+    _new_battery = None
+
+    def __init__(self, hass, lacrosse, device_id, name, expire_after, config):
+        """Initialize the sensor."""
+        self.hass = hass
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, device_id, hass=hass)
+        self._config = config
+        self._name = name
+        self._value = None
+        self._expire_after = expire_after
+        self._expiration_trigger = None
+
+        lacrosse.register_callback(
```

Comment:
It might be safer to move the registering of the callback into the coroutine method `async_added_to_hass` that you can define on the entity. Reason is you need access to the `hass` object in the callback.